vllm/model_executor/models/deepseek_v2.py

```diff
@@ -745,14 +757,49 @@ def load_weights(self, weights: Iterable[tuple[str,
             ("gate_up_proj", "gate_proj", 0),
             ("gate_up_proj", "up_proj", 1),
         ]
-
+        if self.num_share_fusion_replicas > 0:
```

Comment:
Can you put this code in a separate method?   Is it something that might be applied to other models?